packages/core/src/graph/graph.ts

```diff
@@ -1265,16 +1267,95 @@ export default abstract class AbstractGraph extends EventEmitter implements IAbs
     });
 
     //
-    // 2. Add the items to the graph
+    // 2. Compute the origin positions
+    //
+    const existingNodesMap = this.getNodes().reduce((total, node) => {
+      total[node.getID()] = node;
+      return total;
+    }, {});
+    // Map to group the expanded nodes by origin node
+    const originsMap: { [key: string]: { model: NodeConfig, targets: INode[] } } = {};
+    const targetsMap: { [key: string]: { originId: string, pos: { x: number; y: number } } } = {};
+    for (let i = items.length - 1; i > 0; i--) {
```

Comment:
This is because items are reordered so that edges are at the end of the array. In this cycle, I'm iterating from the last item so that as soon as I find something that is not an edge I know that all the edges have been processed